### PR TITLE
added source account and amount validation

### PR DESCRIPTION
**Pull request by Don-PhDev**

### CODE REVIEW

**Review by abe-phdev — APPROVED** [review]

File: app/models/transaction.rb [hunk 1]
```diff
@@ -29,6 +31,21 @@ def actualized?
 
   private
 
+  def source_account_validation
+    is_income = transaction_type_id == TransactionType.income.id
+    is_salary = transaction_type_id == TransactionType.salary.id
+
+    return unless is_income || is_salary
```
Comment (rusbal) [inline app/models/transaction.rb]: Change to 
```
return if transaction_type_id == TransactionType.initialize.id
```

It seems to me that initialization is the only valid transaction that has no source.  Everything else requires a source.

File: app/models/transaction_type.rb [hunk 2]
```diff
@@ -26,6 +26,10 @@ def account_initializer
       first
     end
 
+    def initialize
+      find_by(name: INITIALIZE)
```
Comment (rusbal) [inline app/models/transaction_type.rb]: Change to 
```
@@initialize ||= find_by(name: INITIALIZE)
```

Memoize the following class methods as well.
* `def account_initializer`
* `def income`
* `def salary`

Try to understand what **memoization** means.

File: app/models/transaction_type.rb [hunk 3]
```diff
@@ -26,6 +26,10 @@ def account_initializer
...
+    end
+
     def income
       find_by(name: INCOME_PROGRAMMING)
```
Comment (rusbal) [inline app/models/transaction_type.rb]:
```
@@income ||= find_by(name: INCOME_PROGRAMMING)
```

File: app/models/transaction_type.rb [hunk 4]
```diff
@@ -26,6 +26,10 @@ def account_initializer
       first
```
Comment (rusbal) [inline app/models/transaction_type.rb]:
```
@@account_initializer ||= first
```

File: app/models/transaction_type.rb (hunk 2 above)
Comment (rusbal) [reply app/models/transaction_type.rb]: Type of variables in a class source code

1. `@@i_am_a_class_variable` - Use this when memoizing a class variable
2. `@i_am_an_instance_variable` - Use this when memoizing an instance variable
3. `i_am_a_local_variable` - Cannot be used for memoization

**Review by rusbal — APPROVED** [review]
🚢 :shipit:

**Pull request merged**